components/trustpilot/trustpilot.app.mjs

```diff
@@ -102,479 +88,183 @@ export default {
-        page: Math.floor(offset / limit) + 1,
         language,
-      };
-
-      if (tags.length > 0) {
-        params.tags = tags.join(",");
+        page = 1,
+        internalLocationId,
+        perPage = 20,
+        orderBy = "createdat.desc",
+        tagGroup,
+        tagValue,
+        ignoreTagValueCase = false,
+        responded,
+        referenceId,
+        referralEmail,
+        reported,
+        startDateTime,
+        endDateTime,
+        source,
+        username,
+        findReviewer,
+      } = params;
+
+      // Validate required parameters
+      if (!businessUnitId) {
+        throw new Error("Business Unit ID is required");
       }
-
-      const response = await this._makeRequestWithRetry({
-        endpoint,
-        params,
-      });
-
-      return {
-        reviews: response.reviews?.map(parseReview) || [],
-        pagination: {
-          total: response.pagination?.total || 0,
-          page: response.pagination?.page || 1,
-          perPage: response.pagination?.perPage || limit,
-          hasMore: response.pagination?.hasMore || false,
-        },
-      };
-    },
-
-    async getPublicServiceReviewById({
-      businessUnitId, reviewId,
-    }) {
       if (!validateBusinessUnitId(businessUnitId)) {
-        throw new Error("Invalid business unit ID");
-      }
-      if (!validateReviewId(reviewId)) {
-        throw new Error("Invalid review ID");
+        throw new Error("Invalid business unit ID format");
       }
 
-      const endpoint = buildUrl(ENDPOINTS.PUBLIC_REVIEW_BY_ID, {
+      // Build the endpoint URL
+      const endpoint = buildUrl(ENDPOINTS.SERVICE_REVIEWS, {
         businessUnitId,
-        reviewId,
       });
-      const response = await this._makeRequest({
+
+      // Prepare query parameters
+      const queryParams = {};
+
+      // Add optional parameters if provided
+      if (stars) queryParams.stars = stars;
+      if (language) queryParams.language = language;
+      if (page) queryParams.page = page;
+      if (internalLocationId) queryParams.internalLocationId = internalLocationId;
+      if (perPage) queryParams.perPage = perPage;
+      if (orderBy) queryParams.orderBy = orderBy;
+      if (tagGroup) queryParams.tagGroup = tagGroup;
+      if (tagValue) queryParams.tagValue = tagValue;
+      if (ignoreTagValueCase !== undefined) queryParams.ignoreTagValueCase = ignoreTagValueCase;
+      if (responded !== undefined) queryParams.responded = responded;
+      if (referenceId) queryParams.referenceId = referenceId;
+      if (referralEmail) queryParams.referralEmail = referralEmail;
+      if (reported !== undefined) queryParams.reported = reported;
+      if (startDateTime) queryParams.startDateTime = startDateTime;
+      if (endDateTime) queryParams.endDateTime = endDateTime;
+      if (source) queryParams.source = source;
+      if (username) queryParams.username = username;
+      if (findReviewer) queryParams.findReviewer = findReviewer;
```

Comment:
These checks are not necessary, axios automatically filters out parameters with a value of undefined and doesn't send them in the request.